File: notification-service/src/main/java/org/ricramiel/notificationservice/entity/Notification.java
package org.ricramiel.notificationservice.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "notifications",
        uniqueConstraints = @UniqueConstraint(columnNames = "event_id"))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "event_id", nullable = false, unique = true)
    private UUID eventId;

    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @Column(name = "operation_id")
    private UUID operationId;

    private String type;

    private BigDecimal amount;

    private String currency;

    private String message;

    @Column(name = "is_read")
    @Builder.Default
    private boolean isRead = false;

    @Column(name = "created_at")
    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: front/packages/api/src/entities/notification.ts
/**
 * Соответствует схеме Notification в openapi/notif_api.json (notification-service).
 */
export type Notification = {
  id: string;
  eventId?: string;
  userId?: string;
  operationId?: string;
  type?: string;
  amount?: number;
  currency?: string;
  message?: string;
  createdAt?: string;
  read?: boolean;
};

export function isNotificationPayload(value: unknown): value is Notification {
  if (value === null || typeof value !== "object" || Array.isArray(value)) {
    return false;
  }
  const o = value as Record<string, unknown>;
  return typeof o.id === "string";
}
